contrib/k8s/charts/open-service-broker-azure/templates/deployment.yaml

```diff
@@ -19,6 +19,8 @@ spec:
         chart: "{{ .Chart.Name }}-{{ .Chart.Version }}"
         release: "{{ .Release.Name }}"
         heritage: "{{ .Release.Service }}"
+      annotations:
+        checksum/config: {{ include (print $.Template.BasePath "/broker.yaml") . | sha256sum }}
```

Comment:
One thing I'm not sure: [the doc](https://helm.sh/docs/chart_template_guide/#indenting-and-templates) says `{{ include "TEMPLATE" . }}` will render a template and then place its contents, seems when rendering `broker.yaml`, function `genCA` will be invoked, and a different CA from real rendered `broker.yaml` is used to generate the checksum. I'm not an expert of helm, ping @krancour @zhongyi-zhang , is my assumption correct?

Reply:
OSBA pod must be restarted if certificate changed. It changes every release because it generates.
To track that certificate was changed I have added checksum of broker.yaml to annotation. @norshtein  look please at official helm recommendation [Automatically Roll Deployments When ConfigMaps or Secrets change](https://github.com/helm/helm/blob/master/docs/charts_tips_and_tricks.md#automatically-roll-deployments-when-configmaps-or-secrets-change)

Reply:
yeah, I got your point... what I'm not sure is: here `broker.yaml` is introduced using function `include`, and the doc says function `include` will **render** included file, so when rendering `broker.yaml`, function `genCA` in `broker.yaml` will be called, this generated CA will be different from CA in real `broker.yaml`